doc/src/modules/combinatorics/fp_groups.rst

```diff
@@ -0,0 +1,290 @@
+
+>>> from sympy.combinatorics.free_groups import free_group, vfree_group, xfree_group
+>>> from sympy.combinatorics.fp_groups import FpGroup, CosetTable, coset_enumeration_r
+
+Overview of Facilities
+``````````````````````
+
+The facilities provided for fp-groups fall into a number of natural groupings
+
+* The construction of fp-groups using a free group and a list of words in
+  generators of that free group.
+
+* Index determination using the famous Todd-Coxeter procedure.
+
+* The construction of all subgroups having index less than some (small)
+  specified positive integer, using the *Low-Index Subgroups* algorithm.
+
+* Algorithms for computing presentations of a subgroup of finite index
+  in a group defined by finite presentation.
+
+For a description of fundamental algorithms of finitely presented groups
+we often make use of *Handbook of Computational Group Theory*.
+
+The Construction of Finitely Presented Groups
+---------------------------------------------
+
+Finitely presented groups are construced by factoring a free group by a
+set of relators. The set of relators is taken in as a list of words in
+generators of free group in SymPy, using a list provides ordering to the
+relators. If the list of relators is empty, the associated free group is
+returned.
+
+Example of construction of a finitely-presented group.
+The symmetric group of degree 4 may be represented as a two generator group
+with presentation `< a, b \mid a^2, b^3, (a*b)^4 >`. Giving the relations as a
+list of relators, the presentation of would be specified as:
+
+>>> F, a, b = free_group("a, b")
+>>> G = FpGroup(F, [a**2, b**3, (a*b)**4])
+>>> G
+<fp group on the generators (a, b)>
+
+Currently groups with relators having presentation like
+`< r, s, t \mid r^2, s^2, t^2, rst = str = trs >` will have to be specified as:
+
+>>> F, r, s, t = free_group("r, s, t")
+>>> G = FpGroup(F, [r**2, s**2, t**2, r*s*t*r**-1*t**-1*s**-1, s*t*r*s**-1*r**-1*t**-1])
+
+obviously this is not a unique way to make such a group, but the point is that
+in case of equality with non-identity the user has to manually do that.
+
+Free Groups and Words
+---------------------
+
+Construction of a Free Group
+````````````````````````````
+
+``free_group("gen0, gen1, \ldots gen_(n-1)")`` constructs a free group ``F`` on ``n``
+generators, where ``n`` is a positive integer.
+The `i`-th generator of `F` may be obtained using the method ``.generators[i]``, `i = 0, \ldots n-1`.
+
+>>> F, x, y = free_group("x, y")
+
+creates a free group of rank 2 and assigns the variables ``x`` and ``y`` to the two
+generators.
+
+>>> F = vfree_group("x, y")
+
+creates a free group ``F[0]`` of rank 2, and a tuple of generators ``F[1]``.
+
+
+Construction of words
+`````````````````````
+
+Can be called with different public versions of ``FreeGroup`` i.e
+``free_group``, ``vfree_group`` and ``xfree_group``.
+
+Methods for manipulating Words
+``````````````````````````````
```

Comment:
I am unsure if it would be a good idea to explain each of the methods here. Since we already have the docstring (for each method in in `sympy/combinatorics/free_groups.py`) to address these methods in a more better way.